imnodes.cpp

```diff
@@ -1187,9 +1187,19 @@ void box_selector_update_selection(EditorContext& editor, ImRect box_rect)
     }
 }
 
+void snap_node_to_grid(NodeData &node, float increment)
```

Comment:
With the current code change, the `node_line_handler` function (called by `LoadEditorStateFromIniString`) loads previously stored node coordinates without snapping them. Thus it is possible to have nodes not snapped to a grid, even though `StyleFlags_GridSnapping` is enabled.

We could consider snapping the loaded coordinates to a grid as well, insice `node_line_handler`.

Reply:
I am a little less concerned about nodes being loaded unaligned - the only time I can see that happening is if the user application specifically allows toggling grid snapping on and off, and I'd rather err on the side of loading state as-is, than potentially modifying it in an undesirable way.

However, if this is an important concern I can add this behavior - I leave this at your discretion.